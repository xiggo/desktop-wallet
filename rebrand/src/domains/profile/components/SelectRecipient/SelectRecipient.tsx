import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";
import cn from "classnames";
import React, { useCallback, useEffect, useMemo, useState } from "react";

import { Address } from "@/app/components/Address";
import { Avatar } from "@/app/components/Avatar";
import { Circle } from "@/app/components/Circle";
import { useFormField } from "@/app/components/Form/useFormField";
import { Icon } from "@/app/components/Icon";
import { Select } from "@/app/components/SelectDropdown";
import { TruncateEnd } from "@/app/components/TruncateEnd";
import { useWalletAlias, WalletAliasResult } from "@/app/hooks/use-wallet-alias";
import { AddressProperties, useProfileAddresses } from "@/domains/profile/hooks/use-profile-addresses";
import { SearchRecipient } from "@/domains/transaction/components/SearchRecipient";

type SelectRecipientProperties = {
	network?: Networks.Network;
	address?: string;
	profile: Contracts.IProfile;
	disabled?: boolean;
	isInvalid?: boolean;
	showOptions?: boolean;
	contactSearchTitle?: string;
	contactSearchDescription?: string;
	placeholder?: string;
	exceptMultiSignature?: boolean;
	onChange?: (address: string | undefined, alias: WalletAliasResult) => void;
} & Omit<React.InputHTMLAttributes<any>, "onChange">;

const ProfileAvatar = ({ address }: any) => {
	if (!address) {
		return (
			<Circle
				className="bg-theme-secondary-200 border-theme-secondary-200 dark:bg-theme-secondary-700 dark:border-theme-secondary-700"
				size="sm"
				noShadow
			/>
		);
	}
	return <Avatar address={address} size="sm" noShadow />;
};

const OptionLabel = ({
	option,
	network,
	profile,
}: {
	option: any;
	network?: Networks.Network;
	profile: Contracts.IProfile;
}) => {
	const address = option.value;

	const { getWalletAlias } = useWalletAlias();

	const { alias } = useMemo(
		() =>
			getWalletAlias({
				address,
				network,
				profile,
			}),
		[address, getWalletAlias, network, profile],
	);

	return (
		<div className="flex items-center space-x-2 whitespace-nowrap">
			<Avatar size="sm" address={address} className="flex-shrink-0" noShadow />
			<Address
				address={address}
				walletName={alias}
				addressClass={cn({ "text-theme-primary-600": !alias && option.isSelected })}
				walletNameClass={cn({ "text-theme-primary-600": option.isSelected })}
			/>
		</div>
	);
};

export const SelectRecipient = React.forwardRef<HTMLInputElement, SelectRecipientProperties>(
	(
		{
			address,
			profile,
			disabled,
			isInvalid,
			showOptions = true,
			network,
			placeholder,
			exceptMultiSignature,
			onChange,
		}: SelectRecipientProperties,
		reference,
	) => {
		const { getWalletAlias } = useWalletAlias();

		const [isRecipientSearchOpen, setIsRecipientSearchOpen] = useState(false);

		/*
		 * Initial value for selected address is set in useEffect below via onChangeAddress.
		 * This is made to also retrieve alias information of the selected address.
		 */
		const [selectedAddress, setSelectedAddress] = useState<string | undefined>();
		const [selectedAddressAlias, setSelectedAddressAlias] = useState<WalletAliasResult | undefined>();

		const fieldContext = useFormField();

		const onChangeAddress = useCallback(
			(addressValue: string | undefined, emitOnChange = true) => {
				if (addressValue === selectedAddress) {
					return;
				}

				const alias = getWalletAlias({
					address: addressValue,
					network,
					profile,
				});

				setSelectedAddress(addressValue);
				setSelectedAddressAlias(alias);

				if (emitOnChange) {
					onChange?.(addressValue, alias);
				}
			},
			[getWalletAlias, network, onChange, profile, selectedAddress],
		);

		const isInvalidValue = isInvalid || fieldContext?.isInvalid;

		// Modify the address from parent component
		useEffect(() => {
			onChangeAddress(address, false);
		}, [address]); // eslint-disable-line react-hooks/exhaustive-deps

		const { allAddresses } = useProfileAddresses({ network, profile }, exceptMultiSignature);

		const recipientOptions = allAddresses.map(({ address }: AddressProperties) => ({
			label: address,
			value: address,
		}));

		const openRecipients = () => {
			if (disabled) {
				return;
			}

			setIsRecipientSearchOpen(true);
		};

		const onAction = useCallback(
			(changedAddress: string) => {
				onChangeAddress(changedAddress);
				setIsRecipientSearchOpen(false);
			},
			[onChangeAddress, setIsRecipientSearchOpen],
		);

		return (
			<div>
				<div data-testid="SelectRecipient__wrapper" className="flex relative items-center w-full text-left">
					<Select
						id="SelectRecipient__dropdown"
						showCaret={false}
						isInvalid={isInvalidValue}
						disabled={disabled}
						defaultValue={selectedAddress}
						placeholder={placeholder}
						ref={reference}
						options={showOptions ? recipientOptions : []}
						showOptions={showOptions}
						allowFreeInput={true}
						onChange={(option: any) => onChangeAddress(option.value)}
						addons={{
							end: showOptions
								? {
										content: (
											<div
												data-testid="SelectRecipient__select-recipient"
												className={cn("flex items-center", { "cursor-pointer": !disabled })}
												onClick={openRecipients}
											>
												<Icon name="User" size="lg" />
											</div>
										),
								  }
								: undefined,
							start: {
								content: (
									<div className="flex items-center">
										<ProfileAvatar address={selectedAddress} />
										{!!selectedAddressAlias?.alias && (
											<TruncateEnd
												className="font-semibold ml-2"
												text={selectedAddressAlias.alias}
												showTooltip
											/>
										)}
									</div>
								),
							},
						}}
						renderLabel={(option) => <OptionLabel option={option} network={network} profile={profile} />}
					/>
				</div>

				<SearchRecipient
					isOpen={isRecipientSearchOpen}
					recipients={allAddresses}
					onAction={onAction}
					onClose={() => setIsRecipientSearchOpen(false)}
					selectedAddress={selectedAddress}
				/>
			</div>
		);
	},
);

SelectRecipient.displayName = "SelectRecipient";
