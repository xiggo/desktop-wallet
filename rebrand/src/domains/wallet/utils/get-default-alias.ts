import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
