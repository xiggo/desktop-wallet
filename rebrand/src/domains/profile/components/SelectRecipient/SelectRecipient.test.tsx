/* eslint-disable @typescript-eslint/require-await */
import { Contracts } from "@payvo/sdk-profiles";
import userEvent from "@testing-library/user-event";
import React from "react";

import { SelectRecipient } from "./SelectRecipient";
import { env, getDefaultProfileId, render, screen, waitFor } from "@/utils/testing-library";

let profile: Contracts.IProfile;

const selectRecipient = () => screen.getByTestId("SelectRecipient__select-recipient");

describe("SelectRecipient", () => {
	beforeAll(() => {
		profile = env.profiles().findById(getDefaultProfileId());
	});

	it("should render empty", () => {
		const { container } = render(<SelectRecipient profile={profile} />);

		expect(container).toMatchSnapshot();
	});

	it("should render disabled", () => {
		const { container } = render(<SelectRecipient profile={profile} disabled />);

		expect(container).toMatchSnapshot();
	});

	it("should render invalid", () => {
		const { container } = render(<SelectRecipient profile={profile} isInvalid />);

		expect(container).toMatchSnapshot();
	});

	it("should update internal state when prop changes", () => {
		const { container, rerender } = render(<SelectRecipient profile={profile} />);

		rerender(<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />);

		expect(container).toMatchSnapshot();
	});

	it("should open and close contacts modal", async () => {
		render(<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();

		userEvent.click(selectRecipient());

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("modal__close-btn"));

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();
	});

	it("should select address from contacts modal", async () => {
		render(<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();

		userEvent.click(selectRecipient());

		await expect(screen.findByTestId("modal__inner")).resolves.toBeVisible();

		const firstContactAddress = screen.getByTestId("RecipientListItem__select-button-2");

		userEvent.click(firstContactAddress);

		await waitFor(() => expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument());

		await waitFor(() =>
			expect(screen.getByTestId("SelectDropdown__input")).toHaveValue(
				profile.contacts().values()[0].addresses().values()[0].address(),
			),
		);
	});

	it("should not show select recipient button if showOptions is false", async () => {
		render(<SelectRecipient profile={profile} showOptions={false} />);

		expect(screen.queryByTestId("queryByTestId-recipient")).not.toBeInTheDocument();
	});

	it("should not open contacts modal if disabled", async () => {
		render(<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" disabled />);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();

		userEvent.click(selectRecipient());

		await waitFor(() => expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument());
	});

	it("should call onChange prop when entered address in input", async () => {
		const address = "bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT";

		const contactsSpy = jest.spyOn(profile.contacts(), "findByAddress").mockReturnValue([]);
		const onChange = jest.fn();

		render(<SelectRecipient profile={profile} onChange={onChange} />);
		const recipientInputField = screen.getByTestId("SelectDropdown__input");

		userEvent.paste(recipientInputField, address);

		expect(screen.getByTestId("SelectDropdown__input")).toHaveValue(address);

		expect(onChange).toHaveBeenCalledWith(address, {
			alias: undefined,
			isContact: false,
			isDelegate: false,
		});

		contactsSpy.mockRestore();
	});

	it("should call onChange prop if provided", async () => {
		const onChange = jest.fn();

		render(
			<SelectRecipient
				profile={profile}
				onChange={onChange}
				address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
			/>,
		);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();

		userEvent.click(selectRecipient());

		await waitFor(() => {
			expect(screen.getByTestId("modal__inner")).toBeInTheDocument();
		});

		const firstAddress = screen.getByTestId("RecipientListItem__select-button-2");

		userEvent.click(firstAddress);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();

		const selectedAddressValue = profile.contacts().values()[0].addresses().values()[0].address();

		expect(screen.getByTestId("SelectDropdown__input")).toHaveValue(selectedAddressValue);
		expect(onChange).toHaveBeenCalledWith(selectedAddressValue, expect.any(Object));
	});

	it("should call onChange prop only when values change", async () => {
		const onChange = jest.fn();

		render(<SelectRecipient profile={profile} onChange={onChange} address="D61mfSggzbvQgTUe6JhYKH2doHaqJ3Dyib" />);

		const selectedAddressValue = profile.contacts().values()[0].addresses().values()[0].address();

		expect(screen.getByTestId("SelectDropdown__input")).toHaveValue(selectedAddressValue);

		userEvent.click(selectRecipient());

		await waitFor(() => {
			expect(screen.getByTestId("modal__inner")).toBeInTheDocument();
		});

		const lastAddress = screen.getByTestId("RecipientListItem__selected-button-2");

		userEvent.click(lastAddress);

		expect(screen.getByTestId("SelectDropdown__input")).toHaveValue(selectedAddressValue);
		expect(onChange).not.toHaveBeenCalled();
	});

	it("should filter recipients list by network if provided", async () => {
		const function_ = jest.fn();

		const [wallet] = profile.wallets().findByCoinWithNetwork("ARK", "ark.devnet");

		render(
			<SelectRecipient
				profile={profile}
				onChange={function_}
				address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
				network={wallet.coin().network()}
			/>,
		);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();

		userEvent.click(selectRecipient());

		await waitFor(() => expect(screen.queryByTestId("RecipientListItem__select-button")).not.toBeInTheDocument());
	});

	it("should filter recipients list by MultiSignature type", async () => {
		const { rerender } = render(<SelectRecipient profile={profile} />);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();

		userEvent.click(selectRecipient());

		await waitFor(() => {
			expect(screen.getByTestId("modal__inner")).toBeInTheDocument();
		});

		expect(screen.getAllByTestId("TableRow")).toHaveLength(6);

		const isMultiSignatureSpy = jest
			.spyOn(profile.wallets().first(), "isMultiSignature")
			.mockImplementation(() => true);

		rerender(<SelectRecipient profile={profile} exceptMultiSignature />);

		userEvent.click(selectRecipient());

		await waitFor(() => {
			expect(screen.getByTestId("modal__inner")).toBeInTheDocument();
		});

		expect(screen.getAllByTestId("TableRow")).toHaveLength(1);

		isMultiSignatureSpy.mockRestore();
	});
});
