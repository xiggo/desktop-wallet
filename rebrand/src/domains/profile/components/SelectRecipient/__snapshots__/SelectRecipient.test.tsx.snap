// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectRecipient should render disabled 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="relative w-full"
      >
        <div
          class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR sr-only"
        >
          <div
            class="relative flex flex-1 h-full"
          >
            <input
              class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
              data-testid="select-list__input"
              readonly=""
              tabindex="-1"
              value=""
            />
          </div>
        </div>
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="SelectRecipient__dropdown-menu"
            role="combobox"
          >
            <label
              for="SelectRecipient__dropdown-input"
              id="SelectRecipient__dropdown-label"
            />
            <div
              class="Input__InputWrapperStyled-sc-1i8sd1r-0 ktoJbG"
              disabled=""
            >
              <div
                class="flex items-center"
              >
                <div
                  class="Circle__CircleWrapper-sc-d5u41q-0 bcplCx bg-theme-secondary-200 border-theme-secondary-200 dark:bg-theme-secondary-700 dark:border-theme-secondary-700 ring-theme-background"
                />
              </div>
              <div
                class="relative flex flex-1 h-full"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="SelectRecipient__dropdown-menu"
                  aria-labelledby="SelectRecipient__dropdown-label"
                  autocomplete="off"
                  class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full cursor-default text-theme-secondary-text"
                  data-testid="SelectDropdown__input"
                  disabled=""
                  id="SelectRecipient__dropdown-input"
                  placeholder="Select option"
                  value=""
                />
              </div>
              <div
                class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                data-testid="Input__addon-end"
              >
                <div
                  class=""
                >
                  <div
                    class="flex items-center"
                    data-testid="SelectRecipient__select-recipient"
                  >
                    <div
                      class="sc-bdvvtL qGXhc"
                      height="20"
                      width="20"
                    >
                      <svg>
                        user.svg
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <ul
              aria-labelledby="SelectRecipient__dropdown-label"
              class="styles__SelectOptionsList-sc-120rtxy-0 dJMsqc"
              id="SelectRecipient__dropdown-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectRecipient should render empty 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="relative w-full"
      >
        <div
          class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR sr-only"
        >
          <div
            class="relative flex flex-1 h-full"
          >
            <input
              class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
              data-testid="select-list__input"
              readonly=""
              tabindex="-1"
              value=""
            />
          </div>
        </div>
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="SelectRecipient__dropdown-menu"
            role="combobox"
          >
            <label
              for="SelectRecipient__dropdown-input"
              id="SelectRecipient__dropdown-label"
            />
            <div
              class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="Circle__CircleWrapper-sc-d5u41q-0 bcplCx bg-theme-secondary-200 border-theme-secondary-200 dark:bg-theme-secondary-700 dark:border-theme-secondary-700 ring-theme-background"
                />
              </div>
              <div
                class="relative flex flex-1 h-full"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="SelectRecipient__dropdown-menu"
                  aria-labelledby="SelectRecipient__dropdown-label"
                  autocomplete="off"
                  class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full cursor-default"
                  data-testid="SelectDropdown__input"
                  id="SelectRecipient__dropdown-input"
                  placeholder="Select option"
                  value=""
                />
              </div>
              <div
                class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                data-testid="Input__addon-end"
              >
                <div
                  class=""
                >
                  <div
                    class="flex items-center cursor-pointer"
                    data-testid="SelectRecipient__select-recipient"
                  >
                    <div
                      class="sc-bdvvtL qGXhc"
                      height="20"
                      width="20"
                    >
                      <svg>
                        user.svg
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <ul
              aria-labelledby="SelectRecipient__dropdown-label"
              class="styles__SelectOptionsList-sc-120rtxy-0 dJMsqc"
              id="SelectRecipient__dropdown-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectRecipient should render invalid 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="relative w-full"
      >
        <div
          class="Input__InputWrapperStyled-sc-1i8sd1r-0 lnWRPe sr-only"
        >
          <div
            class="relative flex flex-1 h-full"
          >
            <input
              aria-invalid="true"
              class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
              data-testid="select-list__input"
              readonly=""
              tabindex="-1"
              value=""
            />
          </div>
          <div
            class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-danger-500"
            data-testid="Input__addon-end"
          >
            <span
              data-testid="Input__error"
            >
              <div
                class="sc-bdvvtL qGXhc text-theme-danger-500"
                height="20"
                width="20"
              >
                <svg>
                  circle-exclamation-mark.svg
                </svg>
              </div>
            </span>
          </div>
        </div>
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="SelectRecipient__dropdown-menu"
            role="combobox"
          >
            <label
              for="SelectRecipient__dropdown-input"
              id="SelectRecipient__dropdown-label"
            />
            <div
              class="Input__InputWrapperStyled-sc-1i8sd1r-0 lnWRPe"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="Circle__CircleWrapper-sc-d5u41q-0 bcplCx bg-theme-secondary-200 border-theme-secondary-200 dark:bg-theme-secondary-700 dark:border-theme-secondary-700 ring-theme-background"
                />
              </div>
              <div
                class="relative flex flex-1 h-full"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="SelectRecipient__dropdown-menu"
                  aria-invalid="true"
                  aria-labelledby="SelectRecipient__dropdown-label"
                  autocomplete="off"
                  class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full cursor-default"
                  data-testid="SelectDropdown__input"
                  id="SelectRecipient__dropdown-input"
                  placeholder="Select option"
                  value=""
                />
              </div>
              <div
                class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-danger-500"
                data-testid="Input__addon-end"
              >
                <span
                  data-testid="Input__error"
                >
                  <div
                    class="sc-bdvvtL qGXhc text-theme-danger-500"
                    height="20"
                    width="20"
                  >
                    <svg>
                      circle-exclamation-mark.svg
                    </svg>
                  </div>
                </span>
                <div
                  class="pl-3"
                >
                  <div
                    class="flex items-center cursor-pointer"
                    data-testid="SelectRecipient__select-recipient"
                  >
                    <div
                      class="sc-bdvvtL qGXhc"
                      height="20"
                      width="20"
                    >
                      <svg>
                        user.svg
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <ul
              aria-labelledby="SelectRecipient__dropdown-label"
              class="styles__SelectOptionsList-sc-120rtxy-0 dJMsqc"
              id="SelectRecipient__dropdown-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectRecipient should update internal state when prop changes 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="relative w-full"
      >
        <div
          class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR sr-only"
        >
          <div
            class="relative flex flex-1 h-full"
          >
            <input
              class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
              data-testid="select-list__input"
              readonly=""
              tabindex="-1"
              value="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
            />
          </div>
        </div>
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="SelectRecipient__dropdown-menu"
            role="combobox"
          >
            <label
              for="SelectRecipient__dropdown-input"
              id="SelectRecipient__dropdown-label"
            />
            <div
              class="Input__InputWrapperStyled-sc-1i8sd1r-0 cLNVAR"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="Avatar__AvatarWrapper-sc-18vw8a4-0 jiaHcL flex-shrink-0"
                  data-testid="Avatar"
                >
                  <div
                    class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                  >
                    <img
                      alt="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
                      src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(255, 193, 7)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"20\\" cy=\\"40\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"70\\" fill=\\"rgb(233, 30, 99)\\"/><circle r=\\"55\\" cx=\\"70\\" cy=\\"50\\" fill=\\"rgb(139, 195, 74)\\"/></svg>"
                      title="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
                    />
                  </div>
                </div>
              </div>
              <div
                class="relative flex flex-1 h-full"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="SelectRecipient__dropdown-menu"
                  aria-labelledby="SelectRecipient__dropdown-label"
                  autocomplete="off"
                  class="Input__InputStyled-sc-1i8sd1r-1 hPntwa p-0 border-none bg-transparent focus:ring-0 no-ligatures w-full"
                  data-testid="SelectDropdown__input"
                  id="SelectRecipient__dropdown-input"
                  placeholder="Select option"
                  value="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
                />
              </div>
              <div
                class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                data-testid="Input__addon-end"
              >
                <div
                  class=""
                >
                  <div
                    class="flex items-center cursor-pointer"
                    data-testid="SelectRecipient__select-recipient"
                  >
                    <div
                      class="sc-bdvvtL qGXhc"
                      height="20"
                      width="20"
                    >
                      <svg>
                        user.svg
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <ul
              aria-labelledby="SelectRecipient__dropdown-label"
              class="styles__SelectOptionsList-sc-120rtxy-0 dJMsqc"
              id="SelectRecipient__dropdown-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
